File: C/roman_num_conversion.c
/* Roman numeral conversion:
 * Given a Roman number (I, II, III, IV, ...) in input, convert and print it into arabic numeral system (1, 2, 3, ...)
 * Example:
 *  Input: MMXIX
 *  Output: 2019
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char **argv) {
    char value1[128]; 

    strcpy(value1, argv[1]);    // in this sample code the input is given by terminal (argv[1])

    int convd[strlen(value1)];  // used to convert the roman number to arabic, keeping the order

    // conversion to int
    for(int i = 0; i < strlen(value1); i++){
        switch (value1[i]) {
            case 'I':
                convd[i] = 1;
                break;
            case 'V':
                convd[i] = 5;
                break;
            case 'X':
                convd[i] = 10;
                break;
            case 'L':
                convd[i] = 50;
                break;
            case 'C':
                convd[i] = 100;
                break;
            case 'D':
                convd[i] = 500;
                break;
            case 'M':
                convd[i] = 1000;
                break;
            default:
                break;
        }
    }

    // calculating the sum of the array "convd"

    int sum = 0;

    for(int i = 0; i < strlen(value1); i++){
        if (i == strlen(value1) - 1){
            sum += convd[i];
            break;
        }
        if(convd[i] >= convd[i + 1]){
            sum += convd[i];
        } else {
            sum -= convd[i];
        }
    }

    // now print the result
    printf("%d", sum);
}

/* Bonus exercise: from arabic numeral to Roman numeral */

File: Python/roman_num_conversion.py
"""
Roman numeral conversion:
 * Given a Roman number (I, II, III, IV, ...) in input, convert and print it into arabic numeral system (1, 2, 3, ...)
 * Example:
 *  Input: MMXIX
 *  Output: 2019
"""

import sys

# In this sample code the input is received from terminal (argv[1])
value1 = sys.argv[1]

M = 1000
D = 500
C = 100
L = 50
X = 10
V = 5
I = 1

supp = []
out = 0

for char in value1:
    if char == 'M':
        supp.append(M)
    if char == 'D':
        supp.append(D)
    if char == 'C':
        supp.append(C)
    if char == 'L':
        supp.append(L)
    if char == 'X':
        supp.append(X)
    if char == 'I':
        supp.append(I)
    if char == 'V':
        supp.append(V)

supp.append(0)

# calculate the number in arabic numeral
for i in range(len(supp) - 1):
    if supp[i] < supp[i+1]:
        out -= supp[i]
    else:
        out += supp[i]

print(out)